crates/sui-network/src/state_sync/mod.rs

```diff
@@ -442,89 +442,75 @@ where
+            .digest();
+        if checkpoint.previous_digest != Some(prev_digest) {
+            panic!("Checkpoint {} from consensus has mismatched previous_digest, expected: {:?}, actual: {:?}", checkpoint.sequence_number(), Some(prev_digest), checkpoint.previous_digest);
+        }
 
-        // If this is exactly the next checkpoint then insert it and then notify our peers
-        if *checkpoint.sequence_number() == next_sequence_number
-            && checkpoint.previous_digest == Some(previous_digest)
-        {
-            let checkpoint = *checkpoint;
-
-            // Check invariant that consensus must only send state-sync fully synced checkpoints
-            #[cfg(debug_assertions)]
-            {
-                self.store
-                    .get_full_checkpoint_contents(&checkpoint.content_digest)
-                    .expect("store operation should not fail")
-                    .unwrap();
-            }
+        let latest_checkpoint = self
+            .store
+            .get_highest_verified_checkpoint()
+            .expect("store operation should not fail");
 
-            self.store
-                .insert_checkpoint(checkpoint.clone())
-                .expect("store operation should not fail");
-            self.store
-                .update_highest_synced_checkpoint(&checkpoint)
-                .expect("store operation should not fail");
-            self.metrics
-                .set_highest_verified_checkpoint(*checkpoint.sequence_number());
-            self.metrics
-                .set_highest_synced_checkpoint(*checkpoint.sequence_number());
-
-            // We don't care if no one is listening as this is a broadcast channel
-            let _ = self.checkpoint_event_sender.send(checkpoint.clone());
-
-            self.spawn_notify_peers_of_checkpoint(checkpoint);
-        } else {
-            // Ensure that if consensus sends us a checkpoint that we expect to be the next one,
-            // that it isn't on a fork
-            if *checkpoint.sequence_number() == next_sequence_number {
-                panic!(
-                    "consensus sent a forked checkpoint with different previous_digest, seq_num: {}, expected_previous_digest: {:?}, actual_previous_digest: {:?}",
-                    next_sequence_number, Some(previous_digest), checkpoint.previous_digest
-                );
-            }
+        // If this is an older checkpoint, just ignore it
+        if latest_checkpoint.sequence_number() >= checkpoint.sequence_number() {
+            return;
+        }
 
+        let checkpoint = *checkpoint;
+        let next_sequence_number = latest_checkpoint.sequence_number().saturating_add(1);
+        if *checkpoint.sequence_number() > next_sequence_number {
             debug!(
                 "consensus sent too new of a checkpoint, expecting: {}, got: {}",
                 next_sequence_number,
                 checkpoint.sequence_number()
             );
+        }
```

Comment:
Your PR description and the comment below mentions that this shouldn't happen right?

Reply:
right, being a bit nimble here because there could be bugs somewhere that we overlook